src/components/TodoList/TodoList.jsx

```diff
@@ -1,44 +1,93 @@
 import React from 'react';
 import './TodoList.scss';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import { SearchInput } from '../SearchInput/SearchInput';
 
-export const TodoList = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+export class TodoList extends React.Component {
+  state = {
+    search: '',
+    showTodo: 'all',
+  }
+
+  filterBySelect = {
+    all: callback => callback,
+    completed: callback => callback.completed,
+    active: callback => !callback.completed,
+  }
+
+  handleChange = (event) => {
+    const { name, value } = event.target;
+
+    this.setState({
+      [name]: value,
+    });
+  }
+
+  render() {
+    const { todos, selectedUser, selectedUserId } = this.props;
+    const { search, showTodo } = this.state;
+
+    const filterTodos = todos
+      .filter(todo => todo.title.toLowerCase().includes(search.toLowerCase())
+        && this.filterBySelect[showTodo](todo));
```

Comment:
let's move all the filtering logic to App component, Todolist will accept and render already filtered todos